docker/ci/Dockerfile-clang

```diff
@@ -0,0 +1,37 @@
+FROM ubuntu:16.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update -qq && apt-get install -yqq \
+    build-essential \
+    cmake \
+    g++ \
+    wget \
+    python \
+    software-properties-common
+
+RUN wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+
+run apt-add-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-7 main"
+
+ENV BOOST_ROOT=/usr/local
+ADD util/build_prep/bootstrap_boost.sh bootstrap_boost.sh
+RUN apt-get update -qq && apt-get install -yqq \
+    clang-7 lldb-7 lld-7 libc++-7-dev libc++abi-7-dev 
```

Comment:
should we be using such a new version of clang? A minimum requirement version may show more errors.

Reply:
initially tried clang3.8 which is pulled down with clang, discovered needed 3.9 updated to use 3.9 and symlink that instead